libcloud/compute/drivers/outscale.py

```diff
@@ -7948,3 +7949,283 @@ def _to_key_pair(self, key_pair):
+        """
+        Generates a data key using a master key.
+
+        :param      master_key_id: This is the master_key_id we want to use.
+        :type       master_key_id: ``str``
+
+        :param      size: Size of the data between 1 and 1024.
+        :type       size: ``int``
+
+        :param      generated_plaintext: If true, data received won't be encrypted.
+        :type       generated_plaintext: ``bool``
+
+        :param      encryption_context: This is the context you are using.
+        :type       encryption_context: ``dict``
+
+        :param      dry_run: If true, checks whether you have the required
+        permissions to perform the action.
+        :type       dry_run: ``bool``
+
+        :return: a dict containing 'Plaintext' if generate_plaintext is true and `Ciphertext`.
+        :rtype: ``dict`` of ``str``
+        """
+        action = "GenerateDataKey"
+        data = {"MasterKeyId" : master_key_id, "Size": size, "GeneratePlaintext" : generate_plaintext,  "EncryptionContext" : encryption_context, "DryRun": dry_run}
+        response = self._call_api(action, json.dumps(data))
+        return response.json()
+
+    def ex_read_master_keys(
+        self,
+        filters_master_key: dict = dict(),
+        dry_run: bool = False,
+    ):
+        """
+        Read al master keys.
+
+        :param      filters_master_key: This is the master_key_id we want to use.
+        More at https://docs.outscale.com/api#tocsfiltersmasterkey.
+        :type       filters_master_key: ``str``
+
+        :param      dry_run: If true, checks whether you have the required
+        permissions to perform the action.
+        :type       dry_run: ``bool``
+
+        :return: a dict containing 'Plaintext' if generate_plaintext is true and `Ciphertext`.
+        :rtype: ``dict``
+        """
+        action = "ReadMasterKeys"
+        data = {"DryRun": dry_run, "Filters" : filters_master_key}
+        response = self._call_api(action, json.dumps(data))
+        return response.json()
+
+    def ex_undelete_master_key(
+        self,
+        master_key_id : str,
+        dry_run: bool = False,
+    ):
+        """
+        Abort the deletion of a Customer Master Key.
+
+        :param      master_key_id: This is the master_key_id of the key you are
+        saving from deletion.
+        :type       master_key_id: ``str``
+
+        :param      dry_run: If true, checks whether you have the required
+        permissions to perform the action.
+        :type       dry_run: ``bool``
+
+        :return: True is successfull and dict if error.
+        :rtype: ``dict``
+        """
+        action = "UndeleteMasterKey"
+        data = {"MasterKeyId" : master_key_id, "DryRun": dry_run}
+        response = self._call_api(action, json.dumps(data))
+        if response.status_code == 200:
+            return response.json()["MasterKey"]
+        return response.json()
+
+    def ex_update_master_key(
+        self,
+        master_key_id: str,
```

Comment:
Once again I personnally have a preference for named parameter as it is more readable by the library user but it can be disscussed and I don't think it's objectively better so kind of your call here.